### ISSUE TITLE
[Windows 10] Multiple "Tab Stash" folders created during first install

### ISSUE THREAD

**Issue by cottuli** [josh-berry/tab-stash#103]
### System Details

- **OS Version:** Windows 10 (20H2) 
- **Browser Version:** Firefox 84.0.1 (64-bit)
- **Tab Stash Version:** 2.7.1

**Installed Browser Extensions:**
- None (new Firefox profile)

### Describe the Bug
When a Windows 10 user installs the Tab Stash add-on for the first time, Tab Stash always creates two "Tab Stash" folders in the "Other Bookmarks" folder. 

### Steps to Reproduce
1. Start with a new Firefox profile (NB: uninstalling and reinstalling the "Tab Stash" add-on will not reproduce this bug)
2. Open the Firefox Bookmarks Library in a new window and select the "Other Bookmarks" folder
3. Install the Tab Stash add-on from addons.mozilla.org  
4. Observe that two new "Tab Stash" folders are created in the "Other Bookmarks" folder and (if the user ticks "Allow this extension to run in Private Windows") they are immediately warned that they have ["multiple 'Tab Stash' bookmark folders, and Tab Stash isn't sure which one to use."](https://github.com/josh-berry/tab-stash/wiki/Problems-Locating-the-Tab-Stash-Bookmark-Folder) If the user doesn't check "Allow this extension to run in Private Windows" no warning is given until the user closes and reopens the Tab Stash sidebar menu.

### Impact
This bug affects all new users who are using Windows 10.

### Additional Details
This bug doesn't seem to occur on macOS (tested on macOS Catalina 10.15.6).

<hr>
<em>Readers: If you are also experiencing this bug, please vote for it by giving the ORIGINAL POST a thumbs-up using the :smiley: button.  You are welcome to leave comments and discuss the issue if you have more details to add, but "Me too!" comments are not counted by the voting system.</em>

**Comment by josh-berry** [comment]
Hi, thanks for the report and sorry for the inconvenience!

I was able to reproduce the issue on my side but it took a few tries (it didn't always happen for me). Here's what I think might be happening: both the sidebar page (which is opened automatically when the browser is installed) and the background page (which is always loaded) are checking to see if a "Tab Stash" folder exists. But these two pages run in parallel, so if the timing works out, it's possible that *both* of them will see that the folder doesn't exist, *both* will then try to create it, and you'll get two folders.

Incidentally, this could happen on any platform (not just Windows), although I've observed that for whatever reason, the Firefox bookmarks API is generally more performant on Windows, so it could be that you're just more likely to hit it there.

I'll try to address this soon. In the meantime, there's no harm if you just delete the (empty) duplicate folder.
